lib/net.js

```diff
@@ -468,12 +469,11 @@ Socket.prototype._destroy = function(exception, cb) {
     return;
   }
 
-  self._connecting = false;
-
-  this.readable = this.writable = false;
-
-  for (var s = this; s !== null; s = s._parent)
-    timers.unenroll(s);
+  for (var s = this; s !== null; s = s._parent){
+     timers.unenroll(s);
+     s._connecting = false;
+     s.readable = s.writable = false;
```

Comment:
If we are writing like this, why not just

    s._connecting = s.readable = s.writable = false;

then?